google/cloud/storage/internal/object_write_streambuf.cc

```diff
@@ -142,26 +142,27 @@ void ObjectWriteStreambuf::FlushFinal() {
 
   // Calculate the portion of the buffer that needs to be uploaded, if any.
   auto const actual_size = put_area_size();
-  hash_function_->Update(absl::string_view{pbase(), actual_size});
 
   // After this point the session will be closed, and no more calls to the hash
   // function are possible.
-  auto function = std::move(hash_function_);
-  hash_values_ = std::move(*function).Finish();
   auto upload_request = UploadChunkRequest(upload_id_, committed_size_,
                                            {ConstBuffer(pbase(), actual_size)},
-                                           Merge(known_hashes_, hash_values_));
+                                           hash_function_, known_hashes_);
   request_.ForEachOption(internal::CopyCommonOptions(upload_request));
   OptionsSpan const span(span_options_);
   auto response = client_->UploadChunk(upload_request);
   if (!response) {
     last_status_ = std::move(response).status();
-  } else {
-    committed_size_ = response->committed_size.value_or(0);
-    metadata_ = std::move(response->payload);
-    headers_ = std::move(response->request_metadata);
+    return;
```

Comment:
Nit: Change `if (!B) S1; else return; S2;` to `if (B) return; S1; S2;`.

Reply:
I am not sure I follow, the new code reads:

```cc
if (!B) {
  S1;
  return;
}

S2;
S3;
S4
```

Something like `if (B) return;` is not semantically equivalent.  I would need to write:

```cc
if (B) {
  S2;
  S3;
  S4;
  return;
}

S1;
```

which seems less readable to me.

Reply:
Sorry ... it seems I can't read a diff correctly.